src/ResourceManager/Websites/Commands.Websites/Cmdlets/DeploymentSlots/SwapAzureWebAppSlots.cs

```diff
@@ -0,0 +1,114 @@
+﻿
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+
+using Microsoft.Azure.Commands.WebApps.Utilities;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.WebApps.Cmdlets.DeploymentSlots
+{
+    /// <summary>
+    /// this commandlet will let you swap two web app slots using ARM APIs
+    /// </summary>
+    [Cmdlet("Swap", "AzureRmWebAppSlots", SupportsShouldProcess = true)]
+    public class SwapAzureWebAppSlots : WebAppBaseCmdlet
+    {
+        [Parameter(Position = 0, Mandatory = true, HelpMessage = "Name of the source slot.", ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string SourceSlotName { get; set; }
+
+        [Parameter(Position = 1, Mandatory = false, HelpMessage = "Name of the destination slot.")]
+        [ValidateNotNullOrEmpty]
+        public string DestinationSlotName { get; set; }
+
+        [Parameter(Position = 2, Mandatory = false, HelpMessage = "Swap with preview action to use", ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public SwapWithPreviewAction? SwapWithPreviewAction { get; set; }
+
+        [Parameter(Position = 3, Mandatory = false, HelpMessage = "Flag to determine if VNet should be preserved", ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public bool? PreserveVnet { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Don't ask for confirmation.")]
```

Comment:
Normally you don't want a Force parameter unless there is a special condition you want to query the user to confirm, for example, if you were destroyign existing data as part of this swap.  In this case, you would condition the ShouldContinue prompt on detecting that thsi extra destructive action would occur.

As implemented, this cmdlet does not need a Force parameter.

Note that, the current implementation of ConfirmAction is incorrect, so this it will *require* you to have a Force parameter.  You can get around this by passing 'true' into the first parameter of ConfirmAction, or waiting for my PR later today that changes SHouldProcess implementation.